core/cc/socket_connection.cpp

```diff
@@ -69,19 +69,23 @@ void close(int fd) {
 
 size_t recv(int sockfd, void* buf, size_t len, int flags) {
 #if TARGET_OS == GAPID_OS_WINDOWS
-  return ::recv(sockfd, static_cast<char*>(buf), static_cast<int>(len), flags);
+  auto result =
+      ::recv(sockfd, static_cast<char*>(buf), static_cast<int>(len), flags);
+  return static_cast<size_t>(result > 0 ? result : 0);
 #else   // TARGET_OS == GAPID_OS_WINDOWS
-  return ::recv(sockfd, buf, len, flags);
+  auto result = ::recv(sockfd, buf, len, flags);
+  return static_cast<size_t>(result > 0 ? result : 0);
```

Comment:
`static_cast<size_t>(result > 0 ? result : 0);` Probably deserves to be a method at this point.

Reply:
There is already a method which would be std::max(result, 0), but with all the necessary casting it would even get more complex.

Reply:
```
template<typename T>
size_t clamp_size_t(T val) {
   return static_cast<size_t>(val > 0? val: 0);
}
```

Reply:
Ok, I am convinced :)